File: update_10w.py
#!/usr/bin/env python3

import sys
import time
import pymysql

mysql_server = '172.19.136.33'
mysql_username = 'admin'
mysql_password = 'hechunyang'
mysql_database = 'test'
mysql_port = 3307
mysql_table = 'sbtest1'

# 设置循环分批次批量更改数据N行记录
limit_rows = 100000

db = pymysql.connect(host = mysql_server,
                     user = mysql_username,
                     password = mysql_password,
                     database = mysql_database,
                     port = mysql_port,
                     charset = 'utf8')

cursor = db.cursor()

###################以下代码不用修改##########################
i = 0
lines = []
with open('{0}.txt'.format(mysql_table),'r',encoding="UTF-8" ) as f:
    for line in f:
        i+=1
        lines.append(line.strip())
        if i >= limit_rows:
            linestr = ','.join(lines)
            #print('UPDATE {0} SET pad=\'hechunyang\' WHERE id IN({1})' .format(mysql_table,linestr))
            try:
                # 修改成你的SQL
                rows = cursor.execute('UPDATE {0} SET pad=\'hechunyang\' WHERE id IN({1})' .format(mysql_table,linestr))
                db.commit()
                print('{0}表更改{1}行记录成功.' .format(mysql_table,rows))
                print('-' * 55)
                time.sleep(1)
            except pymysql.Error as e:
                print("Error %d: %s" % (e.args[0], e.args[1]))
                db.rollback()
                sys.exit(0)
            i = 0
            lines.clear()
    db.close()
	
print('{0}表数据更改成功.' .format(mysql_table))

File: update_10w.php
<?php

/*
第一步，得到起始ID
select id,create_time from t1 WHERE create_time>'2022-04-28 00:00:00' and  create_time<'2022-04-29 00:00:00' order by create_time asc limit 10;

第二步，得到结尾ID
select id,create_time from t1 WHERE create_time>'2022-04-28 00:00:00' and  create_time<'2022-04-29 00:00:00' order by create_time desc limit 10;

业务方SQL：
UPDATE t1 SET report_status=0 WHERE create_time>'2022-04-28 00:00:00' and  create_time<'2022-04-29 00:00:00';
*/

#起始ID
$begin_Id = 3188708841;

#结尾ID
$max_Id = 3207131498;

$mysql_server='192.168.32.67';
$mysql_username='yourname'; 
$mysql_password='yourpasswd';
$mysql_database='yourdb';
$mysql_port='3306';
$mysql_table='t1';
###############################################

$conn=mysqli_connect($mysql_server,$mysql_username,$mysql_password,$mysql_database,$mysql_port) or die("error connecting");

if (!$conn){
	die("连接错误: " . mysqli_connect_error());
}

mysqli_query($conn,"set names 'utf8'"); 


while(1==1){
$update = "UPDATE t1 SET report_status=0 WHERE id>=".$begin_Id." AND id<=".($begin_Id=$begin_Id+100000)." LIMIT 100000";
echo $update . PHP_EOL;

mysqli_query($conn,"SET tx_isolation = 'REPEATABLE-READ'");

$result = mysqli_query($conn,$update);

if ($result) {
    if(mysqli_affected_rows($conn)>=1){
            echo "". PHP_EOL;
	echo "t1表更改的行数是: " . mysqli_affected_rows($conn) . PHP_EOL;
	sleep(1);
    }
    else if($begin_Id<$max_Id){
	continue;
    }
   else{
	echo "t1表数据更改成功". PHP_EOL;
	break;
   }
}

}

?>
